exchangelib/properties.py

```diff
@@ -1906,17 +1894,14 @@ class TimeZoneDefinition(EWSElement):
     def from_xml(cls, elem, account):
         return super().from_xml(elem, account)
 
-    def _get_standard_period(self, for_year):
-        # Look through periods and pick a relevant period according to the 'for_year' value
+    def _get_standard_period(self, transitions_group):
+        # Find the first standard period referenced from transitions_group
         valid_period = None
-        for period in sorted(self.periods, key=lambda p: (p.year, p.type)):
-            if period.year > for_year:
+        standard_periods_map = {p.id: p for p in self.periods if p.name == "Standard"}
+        for transition in transitions_group.transitions:
```

Comment:
Suggestion for a simpler logic:

```python
    def _get_standard_period(self, transitions_group):
        # Find the first standard period referenced from transitions_group
        standard_periods_map = {p.id: p for p in self.periods if p.name == "Standard"}
        for transition in transitions_group.transitions:
            try:
                return standard_periods_map[transition.to]
            except KeyError:
                continue
        raise TimezoneDefinitionInvalidForYear(f"Year {for_year} not included in periods {self.periods}")
```

Reply:
What about:
``` python
    def _get_standard_period(self, transitions_group, for_year):
        # Find the first standard period referenced from transitions_group
        standard_periods_map = {p.id: p for p in self.periods if p.name == "Standard"}
        standard_transitions = (t for t in transitions_group.transitions if t.to in standard_periods_map)
        try:
            return next(standard_transitions)
        except StopIteration:
            raise TimezoneDefinitionInvalidForYear(f"Year {for_year} not included in periods {self.periods}")
```

This way the exception only happens in an exception situation, not whenever we're dealing with a `Daylight` transition.

Reply:
A transition group will contain at most two transitions, so performance is not a concern. I prefer readability, and the for loop does that for me, at least.

Reply:
Sure, I have no issue with that. I'll include that in my next change as well.